app/master/cluster_master.py

```diff
@@ -334,3 +357,9 @@ def get_path_for_build_results_archive(self, build_id: int, is_tar_request: bool
             raise ItemNotReadyError('Build artifact file is not yet ready. Try again later.')
 
         return archive_file
+
+    def receive_heartbeat_from_slave(self, slave_id):
+        self._thread_pool_executor.submit(self._async_receive_heartbeat_from_slave(int(slave_id)))
```

Comment:
I don't think this needs to be async, does it? It's only setting a single value. It should only be necessary to spawn a thread if the work triggered by an API call is going to take a significant amount of time.